crates/bevy_render/macros/src/as_bind_group.rs

```diff
@@ -385,5 +393,451 @@ pub fn derive_as_bind_group(input: TokenStream) -> TokenStream {
+    fn parse(input: ParseStream) -> Result<Self> {
+        Ok(Self {
+            lit_int: input.parse()?,
+            _comma: input.parse()?,
+            meta_list: input.parse_terminated(NestedMeta::parse)?,
+        })
+    }
+}
+
+impl Parse for UniformBindingMeta {
+    fn parse(input: ParseStream) -> Result<Self> {
+        Ok(Self {
+            lit_int: input.parse()?,
+            _comma: input.parse()?,
+            ident: input.parse()?,
+        })
+    }
+}
+
+fn get_uniform_binding_attr(attr: &syn::Attribute) -> Result<(u32, Ident)> {
+    let uniform_binding_meta = attr.parse_args_with(UniformBindingMeta::parse)?;
+
+    let binding_index = uniform_binding_meta.lit_int.base10_parse()?;
+    let ident = uniform_binding_meta.ident;
+
+    Ok((binding_index, ident))
+}
+
+fn get_binding_nested_attr(attr: &syn::Attribute) -> Result<(u32, Vec<NestedMeta>)> {
+    let binding_meta = attr.parse_args_with(BindingMeta::parse)?;
+
+    match binding_meta {
+        BindingMeta::IndexOnly(lit_int) => Ok((lit_int.base10_parse()?, Vec::new())),
+        BindingMeta::IndexWithOptions(BindingIndexOptions {
+            lit_int,
+            _comma: _,
+            meta_list,
+        }) => Ok((lit_int.base10_parse()?, meta_list.into_iter().collect())),
+    }
+}
+
+#[derive(Default)]
+enum ShaderStageVisibility {
+    #[default]
+    All,
+    None,
+    Flags(VisibilityFlags),
+}
+
+#[derive(Default)]
+struct VisibilityFlags {
+    vertex: bool,
+    fragment: bool,
+    compute: bool,
+}
+
+impl ToTokens for ShaderStageVisibility {
+    fn to_tokens(&self, tokens: &mut proc_macro2::TokenStream) {
+        tokens.extend(match self {
+            ShaderStageVisibility::All => quote! { ShaderStages::all() },
+            ShaderStageVisibility::None => quote! { ShaderStages::NONE },
+            ShaderStageVisibility::Flags(flags) => {
+                let mut quoted = Vec::new();
+
+                if flags.vertex {
+                    quoted.push(quote! { ShaderStages::VERTEX });
+                }
+                if flags.fragment {
+                    quoted.push(quote! { ShaderStages::FRAGMENT });
+                }
+                if flags.compute {
+                    quoted.push(quote! { ShaderStages::COMPUTE });
+                }
+
+                quote! { #(#quoted)|* }
+            }
+        });
+    }
+}
+
+const VISIBILITY: Symbol = Symbol("visibility");
+const VISIBILITY_VERTEX: Symbol = Symbol("VERTEX");
+const VISIBILITY_FRAGMENT: Symbol = Symbol("FRAGMENT");
+const VISIBILITY_COMPUTE: Symbol = Symbol("COMPUTE");
+const VISIBILITY_ALL: Symbol = Symbol("ALL");
+const VISIBILITY_NONE: Symbol = Symbol("NONE");
+
+fn get_visibility_flag_value(
+    nested_metas: &Punctuated<NestedMeta, Token![,]>,
+) -> Result<ShaderStageVisibility> {
+    let mut visibility = VisibilityFlags::default();
+
+    for meta in nested_metas {
+        use syn::{Meta::Path, NestedMeta::Meta};
+        match meta {
```

Comment:
Is it possible to use either a flag or list syntax for the visibilities? Something like:

- `none`
- `all`
- `vertex`
- `fragment`
- `compute`
- combinations:
  - `vertex|fragment`
  - `vertex|compute`
  - `fragment|compute`
  - `vertex|fragment|compute`
    - same as `all` but maybe someone writes it like this explicitly and I think that is fine

Reply:
Yes. The current changes allow you to do this in a list syntax, for example:

```rust
#[texture(0, visibility(VERTEX, FRAGMENT))]
```

It theoretically would be possible to do it in a flag syntax (like `visibility(vertex|fragment)`) but we would need to provide our own version of `NestedMeta` to do that, given how syn parses the attribute arguments.

Reply:
I think a list would be fine.